go-controller/pkg/node/gateway_shared_intf.go

```diff
@@ -64,71 +65,57 @@ func (npw *nodePortWatcher) AddService(service *kapi.Service) {
 						"%d, stderr: %q, error: %v", npw.gwBridge, svcPort.NodePort, stderr, err)
 				}
 			}
-			// Flows for cloud load balancers on Azure/GCP
-			flows := make([]string, 0)
+		}
+		// Flows for cloud load balancers on Azure/GCP
+		flows := make([]string, 0)
 
-			// Established traffic is handled by default conntrack rules
-			// Table 1 handles forwarding traffic to pods
-			// Table 2 handles return NAT traffic for node port and forwards out of the host
-			// NodePort/Ingress access in the OVS bridge will only ever come from outside of the host
-			for _, ing := range service.Status.LoadBalancer.Ingress {
-				if ing.IP == "" {
-					continue
-				}
-				ingIP := net.ParseIP(ing.IP)
-				if ingIP == nil {
-					klog.Errorf("Failed to parse ingress IP: %s", ing.IP)
-					continue
-				}
-				cookie, err := svcToCookie(service.Name, ingIP, svcPort.Port)
-				if err != nil {
-					klog.Errorf("Unable to generate cookie for svc: %s, %s, %d, error: %v",
-						service.Name, ingIP.String(), svcPort.Port, err)
-					continue
-				}
-				flowProtocol := protocol
-				nw_dst := "nw_dst"
-				nw_src := "nw_src"
-				if utilnet.IsIPv6String(ing.IP) {
-					flowProtocol = protocol6
-					nw_dst = "ipv6_dst"
-					nw_src = "ipv6_src"
-				}
-				// Send to CT from external net with DNAT, eventually will land in table 1 (traffic to pods)
-				flows = append(flows,
-					fmt.Sprintf("cookie=%s,priority=101,in_port=%s,ct_state=-trk,%s,%s=%s,tp_dst=%d,"+
-						"actions=ct(commit,table=0,zone=%d,nat(dst=%s)",
-						cookie, npw.ofportPhys, flowProtocol, nw_dst, ing.IP, svcPort.Port, config.Default.ConntrackZone,
-						util.JoinHostPortInt32(npw.nodeIP.IP.String(), svcPort.NodePort)))
-				// Incoming return traffic from pods, send to CT, unDNAT, goto table 2
-				flows = append(flows,
-					fmt.Sprintf("cookie=%s,priority=101,in_port=%s,%s,%s=%s,tp_src=%d,"+
-						"actions=ct(table=2,zone=%d,nat)",
-						cookie, npw.ofportPatch, flowProtocol, nw_src, npw.nodeIP.IP, svcPort.NodePort, config.Default.ConntrackZone))
+		// Established traffic is handled by default conntrack rules
+		// NodePort/Ingress access in the OVS bridge will only ever come from outside of the host
```

Comment:
@trozet I think we just mean "Ingress" here, not "NodePort/Ingress" since nodeport is certainly accessible on the same host (eg from hostnetwork pods). But you didn't change the comment, so OK.

Reply:
yeah I see the confusion. I meant here "access from the shared bridge" would only come from outside.